### ISSUE TITLE
Significant discrepancy in evaluation metrics compared to the paper

### ISSUE THREAD

**Issue by bbingy** [xianzuwu/Niagara#3]
Hi @xianzuwu ,

Thank you for releasing the code and pretrained models. I tried to run the evaluation following the instructions, but the metrics I obtained differ significantly from those reported in the paper. For example, my PSNR is negative and much lower, and SSIM and LPIPS are also quite off.

Below is a snippet of my evaluation log for reference:

```
➜  Niagara git:(main) ✗ sh evaluate.sh exp/re10k_v2   
/root/miniconda3/envs/flash3d/lib/python3.10/site-packages/requests/__init__.py:86: RequestsDependencyWarning: Unable to find acceptable character detection dependency (chardet or charset_normalizer).
  warnings.warn(
True
[2025-09-03 20:07:10,034][root][INFO] - Current directory: /h20-resource/root/techanah20/Niagara/exp/re10k_v2
[2025-09-03 20:07:10,035][root][INFO] - Working dir: /h20-resource/root/techanah20/Niagara/exp/re10k_v2
/root/miniconda3/envs/flash3d/lib/python3.10/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
/h20-resource/root/techanah20/flash3d/UniDepth-main/unidepth/utils/chamfer_distance.py:9: UserWarning: !! To run evaluation you need KNN. Please compile KNN: `cd unidepth/ops/knn with && bash compile.sh`.
  warnings.warn(
[2025-09-03 20:07:55,019][dinov2][INFO] - using MLP layer as FFN
Not loading pretrained weights for backbone
Load Unidepth--------------------v1---vitl14--------------------
UniDepth_v1_vitl14 is loaded with:
         missing keys: ['pixel_encoder.register_tokens', 'pixel_encoder.norm.weight', 'pixel_encoder.norm.bias']
         additional keys: []
Cannot initialize model with low cpu memory usage because `accelerate` was not found in the environment. Defaulting to `low_cpu_mem_usage=False`. It is strongly recommended to install `accelerate` for faster and less memory-intense model loading. You can do so with: 

pip install accelerate
.
Keyword arguments {'trust_remote_code': True} are not expected by YOSONormalsPipeline and will be ignored.
Loading pipeline components...: 100%|██████████████████████████████████████████████████████████████████████| 6/6 [00:08<00:00,  1.41s/it]
You have disabled the safety checker for <class 'stablenormal.pipeline_yoso_normal.YOSONormalsPipeline'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .
Setting up MemoryEfficientCrossAttention. Query dim is 512, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 512, context_dim is 512 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 512, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 512, context_dim is 512 and using 8 heads.
[2025-09-03 20:08:15,873][root][INFO] - Loading weights from checkpoints/model_re10k_v54.pth...
[2025-09-03 20:08:18,953][root][INFO] - There are 5 test items

100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:10<00:00,  2.08s/it]
[2025-09-03 20:08:29,504][root][INFO] - psnr: -18.03483993858099
[2025-09-03 20:08:29,505][root][INFO] - ssim: 0.00466835904517211
[2025-09-03 20:08:29,505][root][INFO] - lpips: 0.7815169990062714
[2025-09-03 20:08:29,665][root][INFO] - {
    "src": {
        "psnr": -23.25816879272461,
        "ssim": 0.00394984723534435,
        "lpips": 0.7788576126098633
    },
    "tgt5": {
        "psnr": -16.342399048805238,
        "ssim": 0.004332351614721119,
        "lpips": 0.7834130883216858
    },
    "tgt10": {
        "psnr": -16.70790256857872,
        "ssim": 0.004804430273361504,
        "lpips": 0.7828335404396057
    },
    "tgt_rand": {
        "psnr": -15.830889344215393,
        "ssim": 0.005586807057261467,
        "lpips": 0.7809637546539306
    }
}
```

Could you please help me understand what might be causing this discrepancy? Are there any specific environment setups, dependencies, or evaluation settings I might have missed? Additionally, I am attaching the model’s config used during evaluation for your reference.

`{'config': {'exp_name': 'debug', 'file': 'config.yaml'}, 'data_loader': {'batch_size': 1, 'num_workers': 1}, 'train': {'logging': True, 'mixed_precision': '16-mixed', 'num_gpus': [1, 2, 3, 4], 'load_weights_folder': None, 'ema': {'use': True, 'update_every': 10, 'update_after_step': 100, 'beta': 0.9999}, 'scale_pose_by_depth': True, 'use_gt_poses': True}, 'optimiser': {'learning_rate': 0.0001, 'num_epochs': 1, 'scheduler_lambda_step_size': 500000}, 'run': {'resume_ckpt': None, 'dirpath': None, 'debug': False, 'random_seed': 42, 'git_hash': None, 'log_frequency': 250, 'save_frequency': 5000, 'val_frequency': 5000, 'num_keep_ckpts': 5, 'weights_path': 'exp/re10k_v2/checkpoints/model_re10k_v54.pth'}, 'triplane': {'plane_size': 32, 'num_channels': 64}, 'unidepth': {'version': 'v1', 'backbone': 'vitl14'}, 'model': {'attn_heads': 8, 'attn_dim_head': 64, 'attn_layers': 2, 'use_checkpoint': False, 'context_dim': [512, 512], 'encoder': {'out_channels': 512}, 'disable_self_attn': False, 'num_frames': 1, 'depth': {'version': 'v1', 'backbone': 'vitl14'}, 'backbone': {'name': 'resnet', 'num_layers': 50, 'num_ch_dec': [32, 32, 64, 128, 256], 'resnet_bn_order': 'pre_bn', 'weights_init': 'pretrained', 'upsample_mode': 'nearest', 'depth_cond': True}, 'name': 'unidepth', 'frame_ids': [0, -1, 1], 'scales': [0], 'gauss_novel_frames': [-1, 1, 2], 'min_depth': 0.1, 'max_depth': 20, 'gaussians_per_pixel': 2, 'gaussian_rendering': True, 'randomise_bg_colour': True, 'max_sh_degree': 1, 'scaled_offset': False, 'one_gauss_decoder': False, 'predict_offset': True, 'bg_colour': [0.5, 0.5, 0.5], 'shift_rays_half_pixel': 'forward', 'depth_type': 'depth_inc', 'depth_scale': 0.1, 'normal_scale': 0.001, 'xyz_scale': 0.2, 'opacity_scale': 1.0, 'scale_scale': 0.1, 'sh_scale': 1.0, 'scale_lambda': 0.01, 'depth_bias': -0.1, 'normal_bias': 0.0, 'xyz_bias': 0.0, 'opacity_bias': 0.0, 'scale_bias': 0.02, 'renderer_w_pose': True, 'scale_with_depth': False, 'depth_cond': True}, 'experiment': 'layered_re10k', 'eval': {'save_vis': False}, 'dataset': {'name': 're10k', 'split': 'original', 'data_path': '/h20-resource/root/techanah20flash3d/data/RealEstate10K', 'depth_path': None, 'normal_path': '/datasets2/normals', 'unpack_pcl_tar': False, 'preload_depths': False, 'preload_normals': True, 'ransac_on_the_fly': False, 'test_split_path': 'splits/re10k_mine_filtered/test_files.txt', 'height': 256, 'width': 384, 'znear': 0.01, 'zfar': 100.0, 'max_fov': 100.0, 'from_tar': False, 'copy_to_local': True, 'color_aug': False, 'skip_bad_shape': True, 'dilation': 'random', 'max_dilation': 15, 'pad_border_aug': 32, 'subset': -1, 'frame_sampling_method': 'random', 'scale_pose_by_depth': True, 'test_split': 'mine', 'crop_border': True}, 'loss': {'gauss_scale': {'weight': 0.001, 'thresh': 2.0}, 'gauss_offset': {'weight': 0.01, 'thresh': 1.0}, 'mse': {'weight': 1.0, 'type': 'l1'}, 'ssim': {'weight': 0.85}, 'lpips': {'weight': 0.01, 'apply_after_step': 50000}}}`

Thank you very much for your help!
